File: java/src/main/java/com/techelevator/model/ThumbsUpDown.java
package com.techelevator.model;

public class ThumbsUpDown {
    private int thumbsUp;
    private int thumbsDown;
    private int invitationId;
    private int restaurantId;


    public ThumbsUpDown() {
    }

    public ThumbsUpDown(int ThumbsUp, int ThumbsDown, int invitationId, int restaurantId) {
        this.thumbsUp = ThumbsUp;
        this.thumbsDown = ThumbsDown;
        this.invitationId = invitationId;
        this.restaurantId = restaurantId;
    }


    public int getThumbsUp() {
        return this.thumbsUp;
    }

    public void setThumbsUp(int ThumbsUp) {
        this.thumbsUp = ThumbsUp;
    }

    public int getThumbsDown() {
        return this.thumbsDown;
    }

    public void setThumbsDown(int ThumbsDown) {
        this.thumbsDown = ThumbsDown;
    }

    public int getInvitationId() {
        return this.invitationId;
    }

    public void setInvitationId(int invitationId) {
        this.invitationId = invitationId;
    }

    public int getRestaurantId() {
        return this.restaurantId;
    }

    public void setRestaurantId(int restaurantId) {
        this.restaurantId = restaurantId;
    }

    

}

File: frontend/src/Components/ThumbsUpDown/ThumbsUpDown.js
import React, { useState } from 'react'
import { FaThumbsUp, FaThumbsDown } from "react-icons/fa";


export default function ThumbsUpDown(props){
    const restaurantId = props.restaurantId
    const inviteId = props.inviteId

    const [countUp, setCountUp] = useState(0)
    const [countDown, setCountDown] = useState(0)
    const [thumbsUp, setThumbsUp] = useState(false)
    const [thumbsDown, setThumbsDown] = useState(false)

    

    const handleUpClick = () => {
        if(!thumbsUp && thumbsDown){
            setCountUp(countUp + 1)
            setCountDown(countDown - 1)
            setThumbsUp(true)
            setThumbsDown(false)
            props.updateVote(restaurantId,1,-1)
        }
        else if(!thumbsDown && thumbsUp){
            setThumbsUp(false)
            setCountUp(countUp - 1)
            props.updateVote(restaurantId,-1,0)
        }
        else if(!thumbsUp && !thumbsDown){
            setCountUp(countUp + 1)
            setThumbsUp(true)
            props.updateVote(restaurantId,1,0)
        }
      }
    const handleDownClick = () => {
        if(!thumbsDown && thumbsUp){
            setCountDown(countDown + 1)
            setCountUp(countUp - 1)
            setThumbsDown(true)
            setThumbsUp(false)
            props.updateVote(restaurantId,-1,1)
        }
        else if(!thumbsUp && thumbsDown){
            setThumbsDown(false)
            setCountDown(countDown - 1)
            props.updateVote(restaurantId,0,-1)
        }
        else if(!thumbsDown && !thumbsUp){
            setCountDown(countDown + 1)
            setThumbsDown(true)
            props.updateVote(restaurantId,0,1)
        }
    }  

    return(
        <div>
            <button className='thumbs-button' onClick={handleUpClick}><FaThumbsUp />{`${countUp === 0 ? '' : countUp}`}</button>
            <button className='thumbs-button' onClick={handleDownClick}><FaThumbsDown />{`${countDown === 0 ? '' : countDown}`}</button>
        </div>
    )
}